src/test/java/com/kasperin/inventory_management/controllers/v1/ProcessedFoodControllerTest.java

```diff
@@ -0,0 +1,205 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(MockitoExtension.class)
+class ProcessedFoodControllerTest {
+
+    ObjectMapper objectMapper;
+
+    @Mock
+    ProcessedFoodService processedFoodService;
+
+    ProcessedFoodController controller;
+
+    MockMvc mockMvc;
+
+    @BeforeEach
+    void setUp() {
+        objectMapper = new ObjectMapper();
+        controller = new ProcessedFoodController(processedFoodService);
+
+        mockMvc = MockMvcBuilders.standaloneSetup(controller)
+                .setControllerAdvice(new RestResponseEntityExceptionHandler())
+                .build();
+    }
+
+    @Test
+    void findAll() throws Exception {
+        // given
+        ProcessedFood pf1 = new ProcessedFood();
+        pf1.setId(1L);
+        ProcessedFood pf2 = new ProcessedFood();
+        pf2.setId(2L);
+        List<ProcessedFood> processedFoods = Arrays.asList(pf1, pf2);
+
+        when(processedFoodService.findAll()).thenReturn(processedFoods);
+
+        // when
+        mockMvc.perform(
+                get("/api/v1/processedFoods").contentType(MediaType.APPLICATION_JSON))
+                // then
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].id").value(1L))
+                .andExpect(jsonPath("$[1].id").value(2L));
+
+        verify(processedFoodService).findAll();
+        verify(processedFoodService, never()).findByType(any());
+    }
+
+    @Test
+    void findAllByType_vegan() throws Exception {
+        // given
+        ProcessedFood pf = new ProcessedFood();
+        pf.setId(1L);
+        pf.setFoodType(FoodType.VEGAN);
+
+        when(processedFoodService.findByType(any())).thenReturn(Collections.singletonList(pf));
+
+        // when
+        mockMvc.perform(get("/api/v1/processedFoods?type=VEGAN")
+                .accept(MediaType.APPLICATION_JSON))
+                // then
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].foodType").value(FoodType.VEGAN.name()));
+
+        verify(processedFoodService).findByType(eq(FoodType.VEGAN));
+        verify(processedFoodService, never()).findAll();
+    }
+
+    @Test
+    void findAllByType_nonvegan() throws Exception {
```

Comment:
We could parameterize the food type and reuse the test code (see the method above this; these tests are almost identical). But I'm okay with this code duplication. Test code doesn't have to be production-quality